sdk/python/feast/infra/key_encoding_utils.py

```diff
@@ -72,3 +72,30 @@ def serialize_entity_key(
         output.append(val_bytes)
 
     return b"".join(output)
+
+
+def _cast_value_to_str(
```

Comment:
why isn't this just in a dictionary?

Reply:
@franciscojavierarceo 
Could you be more specific about what you mean by "this"?

Reply:
```
mapping = {
    "string_val": (v.string_val, ValueType.STRING),
    "int32_val": str(v.int32_val), ValueType.INT32,
    "int64_val": str(v.int64_val), ValueType.INT64,
}
if value_type not in mapping:
    raise ValueError(f"Value type not supported for plaintext entities: {v}")
else:
    return mapping[value_type]
```

Reply:
lmk if that's more clear

Reply:
What is the benefit of this approach? We're now allocating data on the stack when we probably don't need to.

Reply:
i do not understand what u r saying

Reply:
I'm trying to understand why we would want to dynamically create a dict instead of just checking the cases. Creating the `mapping` dictionary allocates memory on the stack, which is probably more expensive than just the if statements.

Reply:
oh my bad, yeah put that outside the function.

Reply:
```
    mapping = {
        "string_val": (v.string_val, ValueType.STRING),
        "int32_val": str(v.int32_val), ValueType.INT32,
        "int64_val": str(v.int64_val), ValueType.INT64,
    }
def _cast_value_to_str(
    value_type: str,
    v: ValueProto
) -> Tuple[str, int]:
    if value_type not in mapping:
        raise ValueError(f"Value type not supported for plaintext entities: {v}")
    
    return mapping[value_type]
```

Reply:
The mapping depends on `v`, which is only determined at call time.

Reply:
i think the dict is just easier to read was my point but yeah can't have mapping outside of dict duh